crates/tick_probe/src/main.rs

```diff
@@ -24,10 +24,16 @@ fn main() -> Result<()> {
 
     let event = Event::TestTick;
 
-    loop {
+    // Try to prevent drifting away from the given frequency by dynamically computing the sleep duration for each cycle
+    let period = Duration::from_secs_f64(1.0 / args.frequency);
+    let start = Instant::now();
+    for deadline in (0..).map(|i| start + i * period) {
```

Comment:
A bit of paranoia floating point precision, but we've checked on Rust playground that this does make a tiny difference>
```suggestion
    let start = Instant::now();
    for deadline in (0..).map(|i| start + Duration::from_secs_f64(i as f64 / args.frequency) {
```